### DISCUSSION TITLE
How to translate Inchi Keys to Inchi/smiles or mols?

### DISCUSSION THREAD

**Discussion by PascalIversen** [rdkit/rdkit#4671] [Q&A] [answered]
Hey there,
I have a list of molecules for which I only have the Inchi Key. Is there a way to translate the Inchi keys to Inchis, smiles, or rdkit mols?

Thanks!

Best
Pascal

**Comment by dehaenw** [comment, marked as answer]
hello @PascalIversen 
Unfortunately i think that is probably not theoretically possible. If the keys you have correspond to known structures your best bet is a lookup in a database that has both inchi keys and something than can be converted to a mol.
> The nature of the hash algorithm also means that it is virtually impossible to deduce the original InChI from InChIKey (hashes are designed especially for this purpose). The only possible way is to use brute-force method of trying InChIs of all known chemical compounds.
The nature of InChIKey makes it ideal for database storage, especially for indexing purposes. On the other hand it cannot be used as the only format for chemical structure storage because it is not convertible to the original structure.

source: http://inchi.info/inchikey_overview_en.html

**Reply by PascalIversen** [reply]
Makes sense! I'll try to query somewhere.

Thank you for the quick reply!

**Reply by arik-interx** [reply]
pubchem rest server.

**Reply by tomdburns** [reply]
I know this is an old post, but I just stumbled across it, and I just wanted to mention that I've used the requests package to query the pubchem rest server:

url = f"https://pubchem.ncbi.nlm.nih.gov/rest/pug/compound/inchikey/%s/property/InChI/TXT" % inchikey
response = requests.get(__url)
inchi = response.text.strip()

**Reply by Coco501** [reply]
I also wanted to mention CTS-Lite, a tool I've been working on recently. It has 10.6 million compounds from PubChem and can do instant translation between identifiers like this.
